drivers/serial/uart_nrfx_uarte.c

```diff
@@ -524,26 +576,39 @@ static int uarte_instance_init(struct device *dev,
 	return 0;
 }
 
+
 #define UART_NRF_UARTE_DEVICE(idx)					       \
 	DEVICE_DECLARE(uart_nrfx_uarte##idx);				       \
 	UARTE_##idx##_CREATE_TX_BUFF;					       \
 	static struct uarte_nrfx_data uarte_##idx##_data = {		       \
 		UARTE_##idx##_DATA_INIT					       \
```

Comment:
Move this initialization to the end (after `.uart_config`) and please remove this comma from `UARTE_DATA_INT(idx)` definition (we should try to keep all this fancy stuff here consistent).